huaweicloud/services/cce/data_source_huaweicloud_cce_clusters.go

```diff
@@ -0,0 +1,334 @@
+						},
+						"service_network_cidr": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"masters": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"availability_zone": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+								},
+							},
+						},
+						"status": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"endpoints": {
+							Type:     schema.TypeSet,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"url": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"type": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+								},
+							},
+						},
+						"kube_config_raw": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"certificate_clusters": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"name": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"server": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"certificate_authority_data": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+								},
+							},
+						},
+						"certificate_users": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"name": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"client_certificate_data": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"client_key_data": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceCCEClustersV3Read(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	cceClient, err := config.CceV3Client(config.GetRegion(d))
+	if err != nil {
+		return fmtp.DiagErrorf("Unable to create HuaweiCloud CCE client : %s", err)
+	}
+
+	listOpts := clusters.ListOpts{
+		ID:                  d.Get("cluster_id").(string),
+		Name:                d.Get("name").(string),
+		Type:                d.Get("cluster_type").(string),
+		Phase:               d.Get("status").(string),
+		VpcID:               d.Get("vpc_id").(string),
+		EnterpriseProjectID: config.GetEnterpriseProjectID(d),
+	}
+
+	refinedClusters, err := clusters.List(cceClient, listOpts)
+	logp.Printf("[DEBUG] Value of allClusters: %#v", refinedClusters)
+	if err != nil {
+		return fmtp.DiagErrorf("Unable to retrieve clusters: %s", err)
+	}
+
+	ids := make([]string, 0)
```

Comment:
suggest to pre-allocate the memory of ids

Reply:
fixed